### PR TITLE
parsec: fix jinja2 filter filename bug

### PR DESCRIPTION
**Pull request by benfitzpatrick**
If a Jinja2Filter module name ends with `y` or `p` (or `.`) characters then
these characters will be trimmed and an error will be thrown on import. 

This fixes the bug.

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Adding a better solution from Matt...

**Comment by benfitzpatrick** [comment]
...done

File: tests/jinja2/filters/suite.rc-expanded [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Hello from a filter!
+[scheduling]
+    [[dependencies]]
+        graph = foo => bar
+[runtime]
+    [[foo]]
+        pre-command scripting = true
+        command scripting = true
+    [[bar]]
```
Comment (hjoliver) [inline tests/jinja2/filters/suite.rc-expanded]: (github says "No newline at end of file" here!)

File: lib/parsec/Jinja2Support.py [hunk 2]
```diff
@@ -82,7 +82,7 @@ def Jinja2Process( flines, dir, inputs=[], inputs_file=None ):
     for filterdir in usedfdirs:
         sys.path.append( os.path.abspath( filterdir ))
         for f in glob.glob( os.path.join( filterdir, '*.py' )):
-            fname = os.path.basename( f ).rstrip( '.py' )
```
Comment (hjoliver) [inline lib/parsec/Jinja2Support.py]: Oops! (my oops, that is).

**Comment by hjoliver** [comment]
One unrelated minor issue exposed; otherwise looks good, new test passes.  Reassigning to @arjclark for a second review (@cylc/core - we've agreed to double-review all "important" issues, but we might as well do it for everything that is is not obviously trivial - e.g. fixing doc typos - to avoid any mistakes).

**Comment by hjoliver** [comment]
(note the re-assignment was done prior to my email on Friday :grimacing:)

**Comment by arjclark** [comment]
@benfitzpatrick - as per our conversation, can you comment on the "no newline" entry above and change the argument in the jinja2 filter from "boolean" to something else that doesn't look like a type.

**Comment by benfitzpatrick** [comment]
Done.

**Comment by arjclark** [comment]
Looks good to me. Tests battery passing in my environment. Merging in.

**Pull request merged**